File: leetcode/greatestCommonDivisorOfStrings.js
/**
 * @param {string} str1
 * @param {string} str2
 * @return {string}
 */

 function gcd(a, b) {
  if(b === 0) return a
  return gcd(b, a % b)
}

var gcdOfStrings = function(str1, str2) {
  if(!((str1 + str2) === (str2 + str1))) return ""
  let val = gcd(str1.length, str2.length)
  return str1.substring(0, val)
};

File: leetcode/greatestCommonDivisorOfStrings.py
class Solution(object):
    
    def gcd(self, a, b):
        if(b == 0): return a
        return self.gcd(b, a % b)
    
    
    def gcdOfStrings(self, str1, str2):
        """
        :type str1: str
        :type str2: str
        :rtype: str
        """
        if(not((str1 + str2) == (str2 + str1))): return ""
        val = self.gcd(len(str1), len(str2))
        return str1[:val]